[2C-Bug-03]: MainActivity.onDestroy declared protected; BridgeActivity override requires public

**Type:** Bug

**Ticket context:** [2C-07] PR #41 (merged at 0c4b8fb).

**Description:**
The lifecycle hook added to `MainActivity.java` for the BridgeHolder attach/detach pattern declares `onDestroy()` as `protected`, but `com.getcapacitor.BridgeActivity.onDestroy()` is `public`. javac rejects the narrower override and the `:app:compileDebugJavaWithJavac` task fails on the Dev Release workflow against develop.

**Steps to reproduce:**
1. Pull `develop` at 0c4b8fb (or any commit including PR #41).
2. Run `:app:compileDebugJavaWithJavac` (Dev Release workflow does this on push to develop).

**Expected behavior:**
Compile clean.

**Actual behavior:**
```
android/app/src/main/java/com/fluxmeridian/brain3companion/MainActivity.java:15: error: onDestroy() in MainActivity cannot override onDestroy() in BridgeActivity
    protected void onDestroy() {
                   ^
  attempting to assign weaker access privileges; was public
```

**Severity:** Blocker — Dev Release workflow on develop is failing; v2.0.0 DoD demo path cannot be verified.

**Suggested fix:**
Widen the override to `public void onDestroy()`. One-line change. Caught only by Gradle/javac, not by `npm run lint` or `npm run test.unit`. Same Android-SDK gap on the dev machine that surfaced [2C-Bug-02] — local Gradle is not part of the pre-PR verification suite when JDK + ANDROID_HOME are absent.

--- a/android/app/src/main/java/com/fluxmeridian/brain3companion/MainActivity.java
+++ b/android/app/src/main/java/com/fluxmeridian/brain3companion/MainActivity.java
@@ -12,7 +12,7 @@ public class MainActivity extends BridgeActivity {
     }
 
     @Override
-    protected void onDestroy() {
+    public void onDestroy() {
         BridgeHolder.INSTANCE.detach(getBridge());
         super.onDestroy();
     }

